lib/src/impl/realtime/realtime.dart

```diff
@@ -48,13 +53,25 @@ class Realtime extends PlatformObject implements spec.RealtimeInterface {
   Push push;
 
   @override
-  RealtimeChannels channels;
+  RealtimePlatformChannels channels;
 
   @override
   Future<void> close() async => await invoke(PlatformMethod.closeRealtime);
 
+  bool authCallbackInProgress = false;
+
   @override
-  Future<void> connect() async => await invoke(PlatformMethod.connectRealtime);
+  Future<void> connect() async {
+    bool hasAuthCallback = this.options.authCallback!=null;
+    while (hasAuthCallback && authCallbackInProgress) {
```

Comment:
There really shouldn't be a need for this kind of busy-waiting, like ever.

Reply:
addressed: 34417e964349e49ba2071e2ae93316ecf25a8982